tests/test-cases/Group6-VIC-Machine/6-03-Delete.robot

```diff
@@ -95,3 +104,128 @@ Attach Disks and Delete VCH
 
     ${rc}=  Run And Return Rc  govc datastore.ls -dc=%{TEST_DATACENTER} %{VCH-NAME}/VIC/
     Should Be Equal As Integers  ${rc}  1
+
+
+Delete VCH with non-cVM in same RP
+    ${rand}=  Generate Random String  15
+    ${dummyvm}=  Set Variable  anothervm-${rand}
+    Set Suite Variable  ${tempvm}  ${dummyvm}
+    Log To Console  Create VM ${dummyvm} in %{TEST_RESOURCE}/%{VCH-NAME} net %{PUBLIC_NETWORK}
+    ${rc}  ${output}=  Run And Return Rc And Output  govc vm.create -pool="%{TEST_RESOURCE}/%{VCH-NAME}" -net=%{PUBLIC_NETWORK} -on=false ${dummyvm}
+    Should Be Equal As Integers  ${rc}  0
+
+    # Verify VM exists
+    ${rc}  ${output}=  Run And Return Rc And Output  govc ls /ha-datacenter/vm/${dummyvm}
+    Log  ${output}
+    Should Contain  ${output}  ${dummyvm}
+
+    # Delete with force
+    ${ret}  ${output}=  Run And Return Rc And Output  bin/vic-machine-linux delete --target %{TEST_URL} --user %{TEST_USERNAME} --password=%{TEST_PASSWORD} --compute-resource=%{TEST_RESOURCE} --name %{VCH-NAME} --force
+    Log  ${output}
+    Should Contain  ${output}  Completed successfully
+
+    # Verify VM exists
+    ${rc}  ${output}=  Run And Return Rc And Output  govc ls /ha-datacenter/vm/${dummyvm}
+    Log  ${output}
+    Should Contain  ${output}  ${dummyvm}
+
+    # Delete VM and RP
+    ${rc}  ${output}=  Run And Return Rc And Output  govc vm.destroy ${dummyvm}
+    Log  ${output}
+    Should Be Equal As Integers  ${rc}  0
+
+    ${rc}  ${output}=  Run And Return Rc And Output  govc pool.destroy "%{TEST_RESOURCE}/%{VCH-NAME}"
+    Log  ${output}
+    Should Be Equal As Integers  ${rc}  0
+
+    Run Keyword And Ignore Error  Cleanup VCH Bridge Network  %{VCH-NAME}
+
+
+Delete VCH moved from its RP
+    Run Keyword If  '%{HOST_TYPE}' == 'ESXi'  Pass Execution  Test skipped on ESX due to unable to move into RP
```

Comment:
ESX does have resource pools and we should be able to move VMs between them. It's possible that govc is missing exposure of the appropriate command.
(I'm not sure how to view resource pools in the H5 client however).